pkg/controller/helper/shared.go

```diff
@@ -0,0 +1,88 @@
+package helper
+
+import (
+	"context"
+
+	capabilitiesv1alpha1 "github.com/3scale/3scale-operator/apis/capabilities/v1alpha1"
+	
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+)
+
+/*
+RetrieveTenantCR retrieves tenantCR of a tenant that matches the provider account org name
+- providerAccount
+- k8client
+- logger
+If the tenantList is empty it will return nil, nil
+If tenantCR for given providerAccount org is not present, it will return nil, nil
+*/
+func RetrieveTenantCR(providerAccount *ProviderAccount, client k8sclient.Client, logger logr.Logger) (*capabilitiesv1alpha1.Tenant, error) {
+	tenantList := &capabilitiesv1alpha1.TenantList{}
+	err := client.List(context.TODO(), tenantList)
```

Comment:
Namespace scope? Owner references can only be set in the same namespace.

Reply:
But tenant CR can essentially be created under any namespace, this means to me that if a tenant is created via CR, the secret coming from the tenant creation can have a different ns that the tenant CR.
The product and backend must be under same ns as the secret but we still need to loop through all the tenants CRs.

Reply:
Hey, @eguzki as we have agreed on our recent call - going to get the tenant's CRs that are under the same ns as the product or backend. Although this is bringing a bit of limitation on the cascading deletion it looks like the best way to go with it.

Thank you.

Reply:
Hey, @eguzki as we have agreed on our recent call - going to get the tenant's CRs that are under the same ns as the product or backend. Although this is bringing a bit of limitation on the cascading deletion it looks like the best way to go with it.

Thank you.